archaius-core/src/main/java/com/netflix/config/PropertyWrapper.java

```diff
@@ -41,12 +41,7 @@
     private final List<Runnable> callbackList = Lists.newArrayList();
 
     static {
-        PropertyWrapper.registerSubClassWithNoCallback(DynamicIntProperty.class);
```

Comment:
Although creating alternate cached properties would work, I'm concerned that few teams will make the change to use them. And for large systems with many libraries using the existing properties, it ideally would be the choice of the system owners to change this globally for their systems, rather than have to get each library owner to change it in turn.

Taking a look at a large system in prod, jmap shows me ~8000 instances of Dynamic(Boolean|Long|Int|Float|Double)PropertyS, which if 1 instance of a callback conservatively adds 32 bytes, would imply a memory increase of only 256 kB. This seems a reasonable overhead increase to me.

And the overhead of invoking the callback only happens when a property value is changed, which does not happen frequently, and does not happen on the critical path.

So it's ultimately you call, but I think pro's outweigh the cons for making this change to the existing impls.